src/main/java/org/prebid/server/bidder/consumable/ConsumableBidder.java

```diff
@@ -98,6 +102,23 @@ public Result<List<HttpRequest<ConsumableBidRequest>>> makeHttpRequests(BidReque
             requestBuilder.gdpr(bidGdprBuilder.build());
         }
 
+        Optional.ofNullable(request.getSource())
+                .map(Source::getSchain)
+                .ifPresent(requestBuilder::sChain);
+
+        Optional.ofNullable(request.getRegs())
+                .map(Regs::getCoppa)
+                .ifPresent(coppa -> requestBuilder.coppa(coppa == 1));
+
+        Optional.ofNullable(request.getUser())
+                .ifPresent(requestUser -> requestBuilder.user(new ConsumableUser(null, requestUser.getEids())));
+
+        Optional.ofNullable(request.getSite())
+                .map(Site::getContent)
+                .ifPresentOrElse(requestBuilder::content, () -> Optional.ofNullable(request.getApp())
+                                .map(App::getContent)
+                                .ifPresent(requestBuilder::content));
+
```

Comment:
I suggest refactor the code above in the same style

At least I see that the code above already calls the getters for the fields you're used, so you probably you can simplify null-checks

Also I believe the target approach should be resolving all the fields and then pushing them to the single one builder chain instead of splitting building into pieces (if possible)

Reply:
> to the single one builder chain instead of splitting building into pieces

I strongly disagree. Builder patter here doesn't make sense then, also in current case it is more convenient to use/apply values right after check then pass bunch of them somewhere else. And yes, it is not possible, since a lot of values may be absent and can't be applied int request body.

> At least I see that the code above already calls the getters for the fields you're used, so you probably you can simplify null-checks

Only for User. And I can't say that I can simplify too much.

> I suggest refactor the code above in the same style.
Done.

Reply:
there are the null-checks for the user, the site and the regs (not only for the user)

Reply:
it's easily could be something like this

```
final Site site = request.getSite();
        final Regs regs = request.getRegs();
        final User user = request.getUser();
        final Source source = request.getSource();
        final ConsumableBidRequest request = ConsumableBidRequest.builder()
                .time(Instant.now().getEpochSecond())
                .includePricingData(true)
                .enableBotFiltering(true)
                .parallel(true)
                .referrer(Optional.ofNullable(site).map(Site::getRef).orElse(null))
                .url(Optional.ofNullable(site).map(Site::getPage).orElse(null))
                .gpp(Optional.ofNullable(regs).map(Regs::getGpp).orElse(null))
                .gppSid(Optional.ofNullable(regs).map(Regs::getGppSid).filter(CollectionUtils::isNotEmpty).orElse(null))
                .usPrivacy(Optional.ofNullable(regs).map(Regs::getExt).map(ExtRegs::getUsPrivacy).orElse(null))
                .sChain(Optional.ofNullable(source).map(Source::getSchain).orElse(null))
                .coppa(Optional.ofNullable(regs).map(Regs::getCoppa).map(coppa -> coppa == 1).orElse(null))
                .user(Optional.ofNullable(user).map(u -> new ConsumableUser(null, user.getEids())).orElse(null))
                .content(makeContent(site, app))
                .gdpr(makeGdpr(regs, user))
                //same for the imp fields
                .build();
```

Reply:
@KolomiietsSerhii fyi